pkg/credentials/toolstore.go

```diff
@@ -80,16 +80,10 @@ func (h *toolCredentialStore) GetAll() (map[string]types.AuthConfig, error) {
 			}
 		}
 
-		newCredAddresses[toolNameWithCtx(toolName, ctx)] = val
-		delete(serverAddresses, serverAddress)
-	}
-
-	for serverAddress := range newCredAddresses {
-		ac, err := h.Get(serverAddress)
-		if err != nil {
-			return nil, err
+		result[toolNameWithCtx(toolName, ctx)] = types.AuthConfig{
+			Username:      val,
+			ServerAddress: serverAddress,
 		}
-		result[serverAddress] = ac
```

Comment:
This loop is the main thing I needed to fix. Calling `h.Get` results in the credential getting possibly decrypted. I refactored this so that we no longer call `h.Get` here in the GetAll function, and instead call it in the other file, and only on the credentials that we are actually going to return.